src/command_api.rs

```diff
@@ -43,17 +43,41 @@ pub fn run_command_api(command: &str, paths: &GlobalPaths) -> Result<()> {
         "Failed to load configuration file while running the getconfig1 API command."
     })?;
 
-    for (key, value) in config_file.data.installed_channels {
-        let curr = match value {
-            JuliaupConfigChannel::SystemChannel {
-                version: fullversion,
-            } => {
-                let (platform, mut version) = parse_versionstring(&fullversion)
+    for (key, value) in &config_file.data.installed_channels {
+        let curr = match &value {
+            JuliaupConfigChannel::AliasChannel { target } => {
+                // For aliases, we need to resolve to the target and get its info
+                // Skip if the target doesn't exist to avoid infinite recursion
+                if let Some(target_channel) = config_file.data.installed_channels.get(target) {
+                    match target_channel {
+                        JuliaupConfigChannel::AliasChannel { .. } => {
+                            // Avoid infinite recursion for alias-to-alias
```

Comment:
Again: I vote we just ban aliases-of-aliases. Makes the code harder to maintain and less robust, and it saves a lazy user a few keystrokes at best